pkg/controller/handlers/retention/lastused.go

```diff
@@ -0,0 +1,27 @@
+package retention
+
+import (
+	"context"
+
+	"github.com/obot-platform/nah/pkg/router"
+	v1 "github.com/obot-platform/obot/pkg/storage/apis/obot.obot.ai/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// SetLastUsedTime sets the last used time for a thread and its parent thread if it has one.
+// It does not call Update() on the thread, so the caller is responsible for calling Update() if needed.
+func SetLastUsedTime(ctx context.Context, c kclient.Client, thread *v1.Thread) error {
+	thread.Status.LastUsedTime = metav1.Now()
+	if thread.Spec.ParentThreadName != "" {
+		var parentThread v1.Thread
+		if err := c.Get(ctx, router.Key(thread.Namespace, thread.Spec.ParentThreadName), &parentThread); err != nil {
+			return err
+		}
+		parentThread.Status.LastUsedTime = thread.Status.LastUsedTime
+		if err := c.Status().Update(ctx, &parentThread); err != nil {
+			return err
+		}
+	}
+	return nil
+}
```

Comment:
I wanted to capture this logic in a function since we use it three times. It feels kinda weird and arbitrary, since the caller has to call `.Status().Update()` for the thread after calling this, but I didn't know of a better way to do it.